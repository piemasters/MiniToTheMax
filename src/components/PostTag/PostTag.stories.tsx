import type { Meta, StoryObj } from '@storybook/react-webpack5';

import { PostTag } from './PostTag';

const meta: Meta<typeof PostTag> = {
  component: PostTag,
  argTypes: {
    name: {
      name: 'name',
      description: 'The text within the tag',
      defaultValue: 'Tag',
      control: { type: 'text' },
    },
    type: {
      name: 'type',
      description: 'Used to determine the base url',
      defaultValue: 'tag',
      control: { type: 'text' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof PostTag>;

export const Primary: Story = {
  args: {
    name: 'Tag',
    type: 'tag',
  },
};
