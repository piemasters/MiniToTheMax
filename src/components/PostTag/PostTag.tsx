import React, { FC } from 'react';
import kebabCase from 'lodash.kebabcase';

import { PageLink } from '../PageLink';

export interface PostTagProps {
  name: string;
  type: string;
}

export const PostTag: FC<PostTagProps> = ({ name, type }) => {
  return (
    <div data-testid="post-tag" className="inline-block">
      <div className="inline-block px-2 py-1 m-1 bg-gray-200 rounded-md cursor-pointer hover:bg-red-500 text-gray-800 hover:text-white! text-xs">
        <PageLink
          key={name}
          to={`/${type}/${kebabCase(name)}/`}
          type={'cover'}
          direction={'up'}
          className="no-underline"
        >
          {name}
        </PageLink>
      </div>
    </div>
  );
};

export default PostTag;
