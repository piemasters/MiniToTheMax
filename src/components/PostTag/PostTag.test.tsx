import { getByTestId, render } from '@testing-library/react';

import { PostTag } from './PostTag';

describe('PostTag', () => {
  test('renders correctly', () => {
    const { container } = render(<PostTag name={'Tag'} type={'tag'} />);

    expect(getByTestId(container as HTMLElement, 'post-tag')).toHaveTextContent(
      'Tag'
    );
    expect(container).toMatchSnapshot();
  });
});
