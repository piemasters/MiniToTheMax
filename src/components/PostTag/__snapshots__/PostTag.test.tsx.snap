// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTag renders correctly 1`] = `
<div>
  <div
    class="inline-block"
    data-testid="post-tag"
  >
    <div
      class="inline-block px-2 py-1 m-1 bg-gray-200 rounded-md cursor-pointer hover:bg-red-500 text-gray-800 hover:text-white! text-xs"
    >
      <a
        class="no-underline"
        data-testid="page-link"
        href="/tag/tag/"
      >
        Tag
      </a>
    </div>
  </div>
</div>
`;
